QRCoder/QRCodeGenerator.cs

```diff
@@ -802,20 +880,20 @@ public static int Score(ref QRCodeData qrCode)
                                 blackModules++;
 
                     var percent = (blackModules / (qrCode.ModuleMatrix.Count * qrCode.ModuleMatrix.Count)) * 100;
-                    var prevMultipleOf5 = Math.Abs((int) Math.Floor(percent/5)*5 - 50)/5;
-                    var nextMultipleOf5 = Math.Abs((int)Math.Floor(percent / 5) * 5 -45)/5;
-                    score4 = Math.Min(prevMultipleOf5, nextMultipleOf5)*10;
+                    var prevMultipleOf5 = Math.Abs((int)Math.Floor(percent / 5) * 5 - 50) / 5;
+                    var nextMultipleOf5 = Math.Abs((int)Math.Floor(percent / 5) * 5 - 45) / 5;
```

Comment:
Instead of `/ 5` that could be done once in `percent` by `* 100 / 5`.

Reply:
I've changed it to all integer arithmetic:

```cs
//=== OLD ===
double blackModules;
var percent = (blackModules / (qrCode.ModuleMatrix.Count * qrCode.ModuleMatrix.Count)) * 100;
var prevMultipleOf5 = Math.Abs((int)Math.Floor(percent / 5) * 5 - 50) / 5;
var nextMultipleOf5 = Math.Abs((int)Math.Floor(percent / 5) * 5 - 45) / 5;

//=== NEW ===
int blackModules;
var percentDiv5 = blackModules * 20 / (qrCode.ModuleMatrix.Count * qrCode.ModuleMatrix.Count);
var prevMultipleOf5 = Math.Abs(percentDiv5 - 10);
var nextMultipleOf5 = Math.Abs(percentDiv5 - 9);
```

@gfoidl  Please help me verify this does exactly what it did before.

Reply:
From strict math I believe there may be some off by 1 errors with this conversion.
But for the use case (blackModules <= qrCode.ModuleMatrix.Count -- is that always true?) it holds, which I verified by [this sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEUCuA7AHwAEAmABgFgAoUgRmuoDNoACACgEs8MXJ8eAvC1oBuXhH4sAPENpkyYgNSK+3AJTUA3tRa6WzKOy49gAGwCGYANYBZCABMcpmAGcWQhSzOXbDp6+khVQwlRW9rO0dnFw0qPRZtOPi9ADdzQwhTewAlVydBFgB5LLZw3yjXNHF+NREdZN00wzwYAHdcl3z3FgA5NtKLCL9oquDa+oaJ5I5Gdhb2vNMeAEIhTJzFjFiG3USd5KJaAE42ABIAIgAxcw5THFgWcx4AIRsEssj/FwBfR7x7FgAYQSwW+53GSX2ukOJ3WHXyEKhehhbHm8KWiKRLBRmIa3ym+KohOoKPO9ggLXBdSoJNoSCM3CqxjURRK5JwZhgXkG5S+TO41XUWim7M5LAADjAoGAYAKhAMfJ9oiwAPTsYIsABUgq2LO1cgUU2MEtgKRs+Q44uchUYAFZujYngALAB0AEFgC42Jx1I6MK7LqYINA2JLpbKeGrbXqWPa4LGyCyo9T4saWggMOalpbrXaHc73Z7vcy/QGgyGwzKBVGY3GWCho6rYym9FMiAB2dji01ZjA5mA221VdOZi1Wgd2iHEmh0hkYflbXr9Y0fYaVFjGsbCyEbgWViMAEQ4KXtQlXFTc2vITbYGu1YxbumN3ZgZrHudPLFLha9++4R5PFh4zkXFdx4Ede37Qd839H9QylKsMAAusjlAttO1DHt3wnIcWAg7DBynIA==) (note: you can change the `100` to higher numbers, but it takes longer then, so for posting the link I kept the smaller number).

If 
> blackModules <= qrCode.ModuleMatrix.Count -- is that always true?

is always true, then you've done a really nice transform 👍🏻

Reply:
This must be always true. blackModules are defined as blocks in ModuleMatrix with bit 1/true. Thus blackModules is a subset of ModuleMatrix and so can't be bigger than ModuleMatrix.Count.